pipenv/cli/command.py

```diff
@@ -762,18 +762,24 @@ def requirements(state, dev=False, dev_only=False, hash=False):
         echo(crayons.normal(" ".join([prefix, package_index["url"]])))
     if not dev_only:
         for req_name, value in lockfile["default"].items():
-            if hash:
-                hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
+            if value.get("editable", False):
+                echo(crayons.normal("-e " + value["path"]))
             else:
-                hashes = []
-            echo(crayons.normal("".join([req_name, value["version"], *hashes])))
+                if hash:
+                    hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
+                else:
```

Comment:
Probably could leave this section and the one below more like it was before to reduce logical complexity of branching factor,
```
                hashes = []
                if hash:
                    hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
```

Or perhaps this would work?:

```
                hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])] if hashes else []
```

Reply:
Sure, nice
I think second suggestion would be better.

Reply:
No,
The first proposal is better logically and shorter :D

Reply:
Or, How about this?
```
            if value.get("editable", False):
                echo(crayons.normal("-e " + value["path"]))
            elif hash:
                hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
                echo(crayons.normal("".join([req_name, value["version"], *hashes])))
            else:
                echo(crayons.normal("".join([req_name, value["version"]])))
```

Reply:
@hoyaaaa that's looks fine, but I am sure that @oz123 would be very happy if you could also convert this logic off of crayons for color and just click.  Here is an example:  https://github.com/pypa/pipenv/pull/5046/files#diff-ef852c4ac364f946819f765a6bc26f04f1b0968f31fc512949a60fa2ab0685e8R903-R910

Reply:
oz123 suggested removing it rather than converting.
Thanks.